qutip/visualization.py

```diff
@@ -1237,11 +1237,20 @@ def plot_expectation_values(results, ylabels=[], title=None, show_legend=False,
 
     n_e_ops = max([len(result.expect) for result in results])
 
-    if not fig or not axes:
-        if not figsize:
-            figsize = (12, 3 * n_e_ops)
-        fig, axes = plt.subplots(n_e_ops, 1, sharex=True,
-                                 figsize=figsize, squeeze=False)
+    if fig is None:
+        if axes is None:
+            fig, axes = plt.subplots(n_e_ops, 1, sharex=True,
+                                     figsize=(12, 3 * n_e_ops), squeeze=False)
+        else:
+            raise ValueError('figure object is necessary')
```

Comment:
The figure is never used, should we accept a list of axes of the right length?

Reply:
I deleted the else part. This function works if only axes are passed to.